forest/benchmarking/observable_estimation.py

```diff
@@ -1100,31 +964,51 @@ def calibrate_observable_estimates(qc: QuantumComputer, expt_results: List[Exper
+    :param symm_type: the type of symmetrization
+
+        * -1 -- exhaustive symmetrization uses every possible combination of flips; this option
+            is the default since it ensures proper calibration and is only exponential in the
+            weight of each observable (rather than the total number of qubits in a program).
+        * 0 -- no symmetrization
+        * 1 -- symmetrization using an OA with strength 1
+        * 2 -- symmetrization using an OA with strength 2
+        * 3 -- symmetrization using an OA with strength 3
+
+        TODO: accomodate calibration for weight > symmetrization strength (symm_type)
+        Currently, the symmetrization type must be at least the maximum weight of any observable
+        estimated and also match the symmetrization type used to estimate the observables.
+
     :param noisy_program: an optional program from which to inherit a noise model; only relevant
         for running on a QVM
+    :param active_reset: whether or not to begin the program by actively resetting. If true,
+        execution of each of the returned programs in a loop on the QPU will generally be faster.
+    :param show_progress_bar: displays a progress bar via tqdm if true.
     :return: a copy of the input results with updated estimates and calibration results.
     """
     observables = [copy(res.setting.observable) for res in expt_results]
     for obs in observables:
         obs.coefficient = 1.
     observables = list(set(observables))  # get unique observables that will need to be calibrated
 
-    programs = [get_calibration_program(obs, noisy_program) for obs in observables]
+    programs = [get_calibration_program(obs, noisy_program, active_reset) for obs in observables]
     meas_qubits = [obs.get_qubits() for obs in observables]
 
-    symm_progs, symm_qubits, flip_array, prog_groups = symmetrization_method(programs, meas_qubits)
-    symm_outputs = _measure_bitstrings(qc, symm_progs, symm_qubits, num_shots)
-    results = consolidate_symmetrization_outputs(symm_outputs, flip_array, prog_groups)
-
-    assert len(results) == len(meas_qubits) == len(observables)
     calibrations = {}
-    for bitarray, meas_qs, obs in zip(results, meas_qubits, observables):
+    for prog, meas_qs, obs in zip(tqdm(programs, disable=not show_progress_bar), meas_qubits,
+                               observables):
+        results = qc.run_symmetrized_readout(prog, num_shots, symm_type, meas_qs)
+
         # Obtain statistics from result of experiment
-        obs_mean, obs_var = shots_to_obs_moments(bitarray, meas_qs, obs)
-        calibrations[obs.operations_as_set()] = (obs_mean, obs_var, len(bitarray))
+        obs_mean, obs_var = shots_to_obs_moments(results, meas_qs, obs)
+        calibrations[obs.operations_as_set()] = (obs_mean, obs_var, len(results))
 
     for expt_result in expt_results:
+        # TODO: allow weight > symm_type
```

Comment:
One possibility is to use exhaustive if weight > strength of current sym_type. 

Realistically the stabilizers are usually weight 4, for codes people care about. (This suggests, as a separate pyQuil PR, that we might consider trying to find a family of strength 4 OAs.) In general but you might find some use in up to weight 6.